README.md

```diff
@@ -621,6 +622,39 @@ api.PathPart = "users/abc";
 await api.GetAsync();
 ```
 
+HTTP Request Message Properties
+-----------------
+
+In very specific cases (i. e. you use custom `DelegatingHandler`), it might be useful to pass any object reference into the handler. In such case `HttpRequestMessage.Properties` can be used.
+This is done by decorating method parameters with `[HttpRequestMessageProperty("key")]`. If key parameter is not specified then the name of the parameter will be used.
+
+If all (or most) of the methods on the interface pass such object you can specify a `[HttpRequestMessageProperty]` property.
+These work in the same way as path parameters, but they're on the level of the entire API. Properties must have both a getter and a setter.
+
+Property keys used at interface method level must be unique, parameter key must not be same as property key.
+
+For example:
+
+```csharp
+public interface ISomeApi
+{
+    [HttpRequestMessageProperty]
+    object AlwaysIncluded { get; set; }
```

Comment:
Can we use types which the user might want to use, e.g. `string` or `int`? Then show casting them back to the right type after accessing `request.Properties`?